### DISCUSSION TITLE
Allow us to set the dev websocket protocol in remix config

### DISCUSSION THREAD

**Discussion by itxch** [remix-run/remix#3707] [Proposals]
At the moment there is no way to force remix to use `wss` or `ws`. This is set automatically by remix. The logic is in the 
`packages/remix-react/components.tsx`, as part of the `LiveReload` component. Specifically on line 1446
```js
let protocol = location.protocol === "https:" ? "wss:" : "ws:";
let host = location.hostname;
let socketPath = protocol + "//" + host + ":" + ${String(port} + "/socket";
```

This is causing some issues for my team's local development setup, as we are using a `NGINX` proxy to allow for HTTPS locally. This means that the web-socket tries to use `wss`. But this doesn't play nice, due to some SSL issues. 

This results in live reload not working. The easiest fix would be to allow us to force remix to use a certain protocol via a config option.

E.g. Adding a `devWebSocketProtocol` option to `remix.config.js`.

I'd be happy to put a PR together that implements this - if this is a good idea :)

Open to other options/etc

**Comment by kiliman** [comment]
The `<LiveReload/>` component is very simple. I just copied it to my own project. That way I can customize it directly. In fact, I modified it so it calls a special Express endpoint to purge the cache, so only purges on change instead of every request.

**Reply by itxch** [reply]
Yeah, thats not a bad idea - I guess that was the point of creating this thread, to see if it's needed by anyone else - so it can be part of the "main" remix <LiveReload/> component
